From 981e7d512927b321b8cbd3bae2c5854b7294da88 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:53:13 -0400
Subject: [tools] perf/tests: No need to set up ref_reloc_sym

Message-id: <1408953210-26343-306-git-send-email-jolsa@redhat.com>
Patchwork-id: 89038
O-Subject: [PATCH RHEL7.1 BZ1133083 305/322] perf tests: No need to set up ref_reloc_sym
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit c080f72753def150993144d755379941f8b14683
Author: Adrian Hunter <adrian.hunter@intel.com>
Date: Wed Jan 29 16:14:42 2014 +0200

description
===========
Now that ref_reloc_sym is set up by machine__create_kernel_maps(), the
"vmlinux symtab matches kallsyms" test does have to do it.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/tests/vmlinux-kallsyms.c b/tools/perf/tests/vmlinux-kallsyms.c
index 2bd13ed..3d90880 100644
--- a/tools/perf/tests/vmlinux-kallsyms.c
+++ b/tools/perf/tests/vmlinux-kallsyms.c
@@ -26,7 +26,6 @@ int test__vmlinux_matches_kallsyms(void)
  struct map *kallsyms_map, *vmlinux_map;
  struct machine kallsyms, vmlinux;
  enum map_type type = MAP__FUNCTION;
- struct ref_reloc_sym ref_reloc_sym = { .name = "_stext", };
  u64 mem_start, mem_end;
 
  /*
@@ -70,14 +69,6 @@ int test__vmlinux_matches_kallsyms(void)
   */
  kallsyms_map = machine__kernel_map(&kallsyms, type);
 
- sym = map__find_symbol_by_name(kallsyms_map, ref_reloc_sym.name, NULL);
- if (sym == NULL) {
-  pr_debug("dso__find_symbol_by_name ");
-  goto out;
- }
-
- ref_reloc_sym.addr = UM(sym->start);
-
  /*
   * Step 5:
   *
@@ -89,7 +80,6 @@ int test__vmlinux_matches_kallsyms(void)
  }
 
  vmlinux_map = machine__kernel_map(&vmlinux, type);
- map__kmap(vmlinux_map)->ref_reloc_sym = &ref_reloc_sym;
 
  /*
   * Step 6:
-- 
1.7.1